src/seedu/addressbook/AddressBook.java

```diff
@@ -1048,8 +1084,8 @@ private static boolean isPersonDataValid(String[] person) {
      * @param name to be validated
      */
     private static boolean isPersonNameValid(String name) {
-        return name.matches("(\\w|\\s)+");  // name is nonempty mixture of alphabets and whitespace
-        //TODO: implement a more permissive validation
+        // name is nonempty mixture of alphabets and whitespace and hyphens
+        return name.matches("(\\w|\\s|-)+");  
```

Comment:
Hmm this doesn't seem to be related to `sort` command? If so, it should have its own PR.